### PR TITLE
fix: date range bug

### PR DESCRIPTION
**Pull request by rephus**
Closes: #4330

### Description:

[Screencast from 01-02-23 12:26:28.webm](https://user-images.githubusercontent.com/1983672/216031370-0ea0ac90-24d4-4440-9187-18dae5480c30.webm)

### CODE REVIEW

File: packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx [hunk 1]
```diff
@@ -266,7 +266,7 @@ const DateFilterInputs = <T extends ConditionalRule = DateFilterRule>(
                         onChange={(
                             range: [Date | null, Date | null] | null,
                         ) => {
-                            if (range && range[0] && range[1]) {
+                            if (range && range[0]) {
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx:269]: The fix was simple. When we select a new `before` that that happens after `after` date. Bluerprint returns the new range[0] but sets range[1] to undefined, which means that on the previous version, we were never "saving" the new `before` date using `onChange` .

This issue was also happening if you set the `after` before the `first` date. That case is also fixed now.

File: packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx [hunk 2]
```diff
@@ -266,7 +266,7 @@ const DateFilterInputs = <T extends ConditionalRule = DateFilterRule>(
                         onChange={(
                             range: [Date | null, Date | null] | null,
                         ) => {
-                            if (range && range[0] && range[1]) {
+                            if (range && (range[0] || range[1])) {
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx:269]: The fix was simple. When we select a new before that that happens after after date. Bluerprint returns the new range[0] but sets range[1] to undefined, which means that on the previous version, we were never "saving" the new before date using onChange .

This issue was also happening if you set the after before the first date. That case is also fixed now.

**Review by ZeRego — APPROVED** [review]

**Review by IrakliJani — APPROVED** [review]

**Comment by ZeRego** [comment]
@rephus when I select just one date, I get this error :'( 

Screenshot 2023-02-01 at 11 48 26

**Review by ZeRego — CHANGES_REQUESTED** [review]
[check previous comment]

**Comment by rephus** [comment]
> @rephus when I select just one date, I get this error :'( perhaps we need to replace that loop in the onChange and set a default value thinking

Fixed by `adding/removing` 1 day to the selected before/after day. Simply using `new Date()` could cause a different error `Overlapping dates`

File: packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx [hunk 3]
```diff
@@ -266,12 +266,25 @@ const DateFilterInputs = <T extends ConditionalRule = DateFilterRule>(
...
-                                        formatDate(value, undefined, false),
-                                    ),
+                                    values: [
+                                        range[0]
+                                            ? formatDate(
+                                                  range[0],
+                                                  undefined,
+                                                  false,
+                                              )
+                                            : moment(range[1]).add(-1, 'days'),
```
Comment (ZeRego) [inline packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx:279]: don't we need to pass the momentjs result to the formatDate() ? 
```
values: [
  formatDate( range[0] || moment(range[1]).add(-1, 'days'), undefined, false),
  formatDate( range[1] || moment(range[0]).add(1, 'days'), undefined, false),
]
```

Comment (rephus) [reply packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx:279]: formatDate takes a `MomentInput` , which can be a `Date`  (range[0] ) or  `Moment` (  moment(range[1]).add(-1, 'days')) 

Are you suggesting to  do `moment(range[1]).add(-1, 'days').toDate()` to always use a Date here ?

Comment (ZeRego) [reply packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx:279]: My concern is that the fallback value is not being formatted like the rest of the values with `formatDate()` and might cause some issues.

**Review by ZeRego — APPROVED** [review]

**Pull request merged**